mention-storer/src/main/java/com/brandwatch/minibcr/mentionstorer/service/MentionFactory.java

```diff
@@ -0,0 +1,20 @@
+package com.brandwatch.minibcr.mentionstorer.service;
+
+import java.util.UUID;
+
+import org.springframework.stereotype.Service;
+
+import com.brandwatch.minibcr.common.model.Mention;
+
+@Service
+public class MentionFactory {
+
+    public com.brandwatch.minibcr.mentionstorer.entity.Mention getMention(Mention mention) {
+        com.brandwatch.minibcr.mentionstorer.entity.Mention mentionEntity = new com.brandwatch.minibcr.mentionstorer.entity.Mention();
+        mentionEntity.setTitle(mention.getResource().getTitle());
+        mentionEntity.setBody(mention.getResource().getText());
+        mentionEntity.setQueryId(mention.getQueryId());
+        mentionEntity.setId(UUID.randomUUID().toString());
+        return mentionEntity;
```

Comment:
I have two questions about this Solr Mention entity. 

1. Couldn't it be immutable?
2. Why does it have fields from the resource (title, text) even though it also has a `Resource` field, and additionally, why is the `Resource` field not populated here?

Reply:
1. Need to check on this as my assumption is setters are being used for data read and write from Solr. Will check with doing private setters but making really immutable by setting final would may not be possible.
2.Fixed